0.10.0版本通过VPN下载番剧失败

通过回国VPN下载番剧失败，有挂自己的大会员cookie：
```
annie -c cookie.txt https://www.bilibili.com/bangumi/play/ep317932
annie -aria2 -c cookie.txt https://www.bilibili.com/bangumi/play/ep317932
```
不管是通过annie本身的downloader还是aria2都不行，退回到0.9.8版本是没有问题的。
之后尝试了用0.10.0版本下载不需要回国VPN的番剧（例：[https://www.bilibili.com/bangumi/play/ep89591](https://www.bilibili.com/bangumi/play/ep89591)），则没有问题。

annie -d的错误相关输出：
```
Downloading https://www.bilibili.com/bangumi/play/ep317932 error:
http://cn-gdfs2-cmcc-v-08.bilivideo.com/upgcxcode/37/11/196401137/196401137-1-112.flv?expires=****&platform=pc&ssig=****&oi=****&trid=****&nfb=****&nfc=****&mid=**** request error: HTTP 454
```
aria2的错误相关输出：
```
2020-05-30 13:41:02.618921 [INFO] [HttpConnection.cc:164] CUID#10 - Response received:
HTTP/1.1 453 
Server: openresty
Date: Sat, 30 May 2020 17:41:02 GMT 
Content-Length: 0
Connection: keep-alive
X-GWF: 1

…

2020-05-30 13:41:02.619034 [ERROR] [AbstractCommand.cc:351] CUID#10 - Download aborted. URI=http://cn-gdfs2-cmcc-v-08.bilivideo.com/upgcxcode/37/11/196401137/196401137-1-112.flv?expires=****&platform=pc&ssig=****&oi=****&trid=****&nfb=****&nfc=****&mid=****
Exception: [AbstractCommand.cc:351] errorCode=22 URI=http://cn-gdfs2-cmcc-v-08.bilivideo.com/upgcxcode/37/11/196401137/196401137-1-112.flv?expires=****&platform=pc&ssig=****&oi=****&trid=****&nfb=****&nfc=****&mid=****
  -> [HttpSkipResponseCommand.cc:240] errorCode=22 The response status is not successful. status=453
```

--- a/downloader/downloader.go
+++ b/downloader/downloader.go
@@ -110,7 +110,7 @@ func (downloader *Downloader) writeFile(url string, file *os.File, headers map[s
 	return written, nil
 }
 
-func (downloader *Downloader) save(part *types.Part, fileName string) error {
+func (downloader *Downloader) save(part *types.Part, refer, fileName string) error {
 	filePath, err := utils.FilePath(fileName, part.Ext, downloader.option.FileNameLength, downloader.option.OutputPath, false)
 	if err != nil {
 		return err
@@ -131,7 +131,9 @@ func (downloader *Downloader) save(part *types.Part, fileName string) error {
 	if err != nil {
 		return err
 	}
-	headers := make(map[string]string, 1)
+	headers := map[string]string{
+		"Referer": refer,
+	}
 	var (
 		file      *os.File
 		fileError error
@@ -206,7 +208,7 @@ func (downloader *Downloader) save(part *types.Part, fileName string) error {
 	return nil
 }
 
-func (downloader *Downloader) multiThreadSave(dataPart *types.Part, fileName string) error {
+func (downloader *Downloader) multiThreadSave(dataPart *types.Part, refer, fileName string) error {
 	filePath, err := utils.FilePath(fileName, dataPart.Ext, downloader.option.FileNameLength, downloader.option.OutputPath, false)
 	if err != nil {
 		return err
@@ -336,7 +338,9 @@ func (downloader *Downloader) multiThreadSave(dataPart *types.Part, fileName str
 			}()
 
 			var end, chunkSize int64
-			headers := make(map[string]string, 1)
+			headers := map[string]string{
+				"Referer": refer,
+			}
 			if downloader.option.ChunkSizeMB <= 0 {
 				chunkSize = part.End - part.Start + 1
 			} else {
@@ -579,9 +583,9 @@ func (downloader *Downloader) Download(data *types.Data) error {
 		// only one fragment
 		var err error
 		if downloader.option.MultiThread {
-			err = downloader.multiThreadSave(stream.Parts[0], title)
+			err = downloader.multiThreadSave(stream.Parts[0], data.URL, title)
 		} else {
-			err = downloader.save(stream.Parts[0], title)
+			err = downloader.save(stream.Parts[0], data.URL, title)
 		}
 
 		if err != nil {
@@ -611,7 +615,7 @@ func (downloader *Downloader) Download(data *types.Data) error {
 		wgp.Add()
 		go func(part *types.Part, fileName string) {
 			defer wgp.Done()
-			err := downloader.save(part, fileName)
+			err := downloader.save(part, data.URL, fileName)
 			if err != nil {
 				lock.Lock()
 				errs = append(errs, err)

--- a/downloader/downloader_test.go
+++ b/downloader/downloader_test.go
@@ -17,6 +17,7 @@ func TestDownload(t *testing.T) {
 				Site:  "douyin",
 				Title: "test",
 				Type:  types.DataTypeVideo,
+				URL:   "https://www.douyin.com",
 				Streams: map[string]*types.Stream{
 					"default": {
 						ID: "default",
@@ -37,6 +38,7 @@ func TestDownload(t *testing.T) {
 				Site:  "douyin",
 				Title: "test2",
 				Type:  types.DataTypeVideo,
+				URL:   "https://www.douyin.com",
 				Streams: map[string]*types.Stream{
 					"miaopai": {
 						ID: "miaopai",
@@ -69,6 +71,7 @@ func TestDownload(t *testing.T) {
 				Site:  "bcy",
 				Title: "bcy image test",
 				Type:  types.DataTypeImage,
+				URL:   "https://www.bcyimg.com",
 				Streams: map[string]*types.Stream{
 					"default": {
 						ID: "default",

--- a/extractors/instagram/instagram_test.go
+++ b/extractors/instagram/instagram_test.go
@@ -17,7 +17,7 @@ func TestDownload(t *testing.T) {
 			args: test.Args{
 				URL:   "https://www.instagram.com/p/BlIka1ZFCNr",
 				Title: "Instagram BlIka1ZFCNr",
-				Size:  3096378,
+				Size:  3003662,
 			},
 		},
 		{

